sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py

```diff
@@ -114,18 +114,21 @@
 TRANSFER_DEAD_LETTER_QUEUE_SUFFIX = '/$Transfer' + DEAD_LETTER_QUEUE_SUFFIX
 
 
-class ReceiveMode(Enum):
-    PeekLock = constants.ReceiverSettleMode.PeekLock
-    ReceiveAndDelete = constants.ReceiverSettleMode.ReceiveAndDelete
+class ServiceBusReceiveMode(Enum):
```

Comment:
I remember in other sdks like schema registry or storage blob, the class would actually inherit from both str and Enum like this:
```suggestion
class ServiceBusReceiveMode(str, Enum):
```

so that we could enable the str comparison:
like 
```python
"PeekLock" == ServiceBusReceiveMode.PeekLock # which would be True
```

Reply:
I have this fixed in a local change, just been waiting to get an OK on the enum naming inconsistencies with MGMT before I push it in, but good call :)

Reply:
See comment below for full details on the resolution, but this is aligned.